crates/multimodal/src/opencv_buffer_capture.cpp

```
@@ -0,0 +1,96 @@
+  long long read(char* buffer, long long size) override {
+    if (size <= 0 || position_ >= size_) {
+      return 0;
+    }
+    const size_t count =
+        std::min(static_cast<size_t>(size), size_ - position_);
+    std::memcpy(buffer, data_ + position_, count);
+    position_ += count;
+    return static_cast<long long>(count);
+  }
+
+  long long seek(long long offset, int origin) override {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)